user-operator/src/main/java/io/strimzi/operator/user/UserOperatorConfig.java

```diff
@@ -364,194 +296,64 @@ private static boolean getBooleanProperty(Map<String, String> map, String name,
-     */
-    public int getScramPasswordLength() {
-        return scramPasswordLength;
-    }
-
-    /**
-     * @return  Indicates whether the Kafka Admin API for managing ACLs is supported by the Kafka cluster or not
-     */
-    public boolean isAclsAdminApiSupported() {
-        return aclsAdminApiSupported;
-    }
-
-    /**
-     * @return  Indicates whether KRaft is used in the Kafka cluster or not. When it is used, some APIs might need to be
-     * disabled or used differently.
-     */
-    public boolean isKraftEnabled() {
-        return kraftEnabled;
-    }
-
-    /**
-     * @return List of maintenance windows. Null if no maintenance windows were specified.
-     */
-    public List<String> getMaintenanceWindows() {
-        return maintenanceWindows;
-    }
-
-    /**
-     * @return Properties object with the user-supplied configuration for the Kafka Admin Client
-     */
-    public Properties getKafkaAdminClientConfiguration() {
-        return kafkaAdminClientConfiguration;
-    }
-
-    /**
-     * @return  The timeout after which operations are considered as failed
-     */
-    public long getOperationTimeoutMs() {
-        return operationTimeoutMs;
-    }
-
-    /**
-     * @return  The size of the User Controller work queue
+     * @return Set of configuration key/names
      */
-    public int getWorkQueueSize() {
-        return workQueueSize;
+    public static Set<String> keyNames() {
+        return Collections.unmodifiableSet(CONFIG_VALUES.keySet());
     }
 
     /**
-     * @return  Size of the pool of the controller threads used to reconcile the users
-     */
-    public int getControllerThreadPoolSize() {
-        return controllerThreadPoolSize;
-    }
-
-    /**
-     * @return  Refresh interval for the cache storing the resources from the Kafka Admin API
+     * Checks if the configuration values are known or not.
+     *
+     * @param map   The map containing configuration values
+     * @param value The configuration value that need to be checked
+     * @return The configuration value
      */
-    public long getCacheRefresh() {
-        return cacheRefresh;
-    }
+    private <T> T get(Map<String, String> map, ConfigParameter<T> value) {
+        if (!CONFIG_VALUES.containsKey(value.key)) {
+            throw new InvalidConfigurationException("Unknown config value: " + value.key + " probably needs to be added to Config.CONFIG_VALUES");
+        }
 
-    /**
-     * @return  Maximal queue for requests when micro-batching the Kafka Admin API requests
-     */
-    public int getBatchQueueSize() {
-        return batchQueueSize;
+        final String s = map.getOrDefault(value.key, value.defaultValue);
+        if (s != null) {
+            if ((value.key.equals(STRIMZI_NAMESPACE) || value.key.equals(STRIMZI_CA_CERT_SECRET_NAME) || value.key.equals(STRIMZI_CA_KEY_SECRET_NAME)) && s.equals("")) {
+                throw new InvalidConfigurationException("Config value: " + value.key + " is mandatory");
```

Comment:
Why do we need to special case these when the `ConfigParameter` already knows about required-ness?

Reply:
The required-ness is fine but  if the the user gives an empty string "" like this then it will be treated as a correct namespace if we don't include this check